### PR TITLE
Github action for cli

### PR DESCRIPTION
**Pull request by mbj36**
Next step

- Add valid s3 link in `package.json`
- Use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment vars.
- create `NPM_TOKEN` and add it to github repo under settings

@gja @akashjain971

### CODE REVIEW

File: .github/workflows/github_actions.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: v ${{ github.ref }}
+          draft: false
+          prerelease: false
+  build:
```
Comment (gja) [inline .github/workflows/github_actions.yml]: job name is the same as previous?

File: .github/workflows/github_actions.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: v ${{ github.ref }}
```
Comment (gja) [inline .github/workflows/github_actions.yml]: should this space be here?

File: .github/workflows/github_actions.yml (hunk 1 above)
Comment (gja) [reply .github/workflows/github_actions.yml]: Also, can't this be the same job, just different steps?

File: .github/workflows/github_actions.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+  build:
+    name: Creating a oclif binary and uploading to S3
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: oclif-dev pack
+      - uses: actions/checkout@v2
+      - run: oclif-dev pack:win
```
Comment (gja) [inline .github/workflows/github_actions.yml]: pack:win and pack:macos won't work on ubuntu

File: .github/workflows/github_actions.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  build:
+    name: Publishing to NPM
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - run: yarn install
```
Comment (gja) [inline .github/workflows/github_actions.yml]: no yarn again

**Review by gja — CHANGES_REQUESTED** [review]

File: .github/workflows/github_actions.yml (hunk 1 above)
Comment (mbj36) [reply .github/workflows/github_actions.yml]: Changed it to different job name. GH was throwing some errors with runs and uses so kept different job

**Pull request merged**